src/fast_scanning.jl

```diff
@@ -0,0 +1,136 @@
+sharpness{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}) = var(imfilter(img, Kernel.Laplacian(ntuple(i->true, Val{N}))))
+
+function adaptive_thres{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}, block::NTuple{N,Int})
```

Comment:
>Do you have a citation for this algorithm?

The algorithm used in `adaptive_thres` has been demonstrated in [this paper](http://djj.ee.ntu.edu.tw/FastSegment.pdf). They have used a linear combination of the variance and frequency as a measure of the threshold of each block. The coefficients have been adjusted to work with common grayscale images.
The sharpness (frequency) of an image is calculated using the variance of Laplacian as [demonstrated here](https://www.researchgate.net/publication/234073157_Analysis_of_focus_measure_operators_in_shape-from-focus) (see LAP4).

>Is it something that might have more widespread use?

The sharpness function might be useful. However, there are multiple more methods for measuring the focus of an image (about 30 methods have been compared in the mentioned paper above). The `adaptive_thres` function uses constants that have been hard-coded to work nicely only with a small class of images and only when using `abs(x1-x2)` as the difference measure. I don't think it has much widespread use.